File: WeSharper-Backend/Models/Friend.cs
using System;
using System.Collections.Generic;

namespace WeSharper.Models
{
    public partial class Friend
    {
        public string RelationshipId { get; set; } = null!;
        public string? RequestedUserId { get; set; }
        public string? AcceptedUserId { get; set; }
        public bool IsAccepted { get; set; }
        public string? Relationship { get; set; }
        public DateTime? CreatedAt { get; set; }

        public virtual ApplicationUser? AcceptedUser { get; set; }
        public virtual ApplicationUser? RequestedUser { get; set; }
    }
}

File: WeSharper-Frontend/Frontend/src/app/_models/friend.ts
import { Profile } from "./profile";

export interface Friend {
    relationshipId: string,
    requestUserId: string,
    acceptedUserId: string,
    isAccepted: boolean,
    createdAt: Date,
    acceptedUser: [
        profiles: Profile
    ],
    requestedUser: [
        profiles: Profile
    ],
}